app/forms/collection_form.rb

```diff
@@ -41,7 +41,10 @@ def build_child
 
   def save
     if valid?
-      run_callbacks(:save) { collection.all?(&:save) }
+      run_callbacks(:save) do
```

Comment:
Does run callbacks treat whatever happens in the block as an atomic action and wrap it in a transaction?